examples/basic/evm-example/src/components/SwapContent/SwapContent.tsx

```diff
@@ -0,0 +1,599 @@
+        toChain?.type === TransactionType.COSMOS) &&
+      !keplrWalletState.connected
+    ) {
+      try {
+        await connect(WalletTypes.KEPLR)
+      } catch (error) {
+        console.log(error)
+
+        return 'Error connecting to Keplr. Please check Keplr and try again.'
+      }
+    }
+
+    if (
+      (fromChain?.type === TransactionType.SOLANA ||
+        toChain?.type === TransactionType.SOLANA) &&
+      !phantomWalletState.connected
+    ) {
+      try {
+        await connect(WalletTypes.PHANTOM)
+      } catch (error) {
+        console.log(error)
+
+        return 'Error connecting to Phantom. Please check Phantom and try again.'
+      }
+    }
+
+    return false
+  }
+
+  const handleSwap = async () => {
+    setError('')
+    setQuote(null)
+
+    if (!fromChain) {
+      setError(`Please select source blockchain.`)
+      return
+    }
+    if (!fromToken) {
+      setError(`Please select source token.`)
+      return
+    }
+    if (!toChain) {
+      setError(`Please select destination blockchain.`)
+      return
+    }
+    if (!toToken) {
+      setError(`Please select destination token.`)
+      return
+    }
+    if (!inputAmount) {
+      setError('Set input amount')
+      return
+    }
+
+    const connectedWalletError = await checkConnectedWalletError()
+
+    if (connectedWalletError) {
+      setError(connectedWalletError)
+      return
+    }
+
+    if (
+      fromChain?.type === TransactionType.EVM &&
+      fromChain?.chainId &&
+      parseInt(fromChain.chainId) !==
+        parseInt(providers()[WalletTypes.META_MASK]?.chainId)
+    ) {
+      try {
+        const provider = providers()[WalletTypes.META_MASK]
+        await provider.send('wallet_switchEthereumChain', [
+          { chainId: `0x${Number(fromChain.chainId).toString(16)}` },
+        ])
+      } catch (e) {
+        setError(`Change meta mask network to '${fromChain?.name}'.`)
+        return
+      }
+    }
+
+    const fromAddress = getAddressByChain(fromChain)
+    const toAddress = getAddressByChain(toChain)
+
+    if (!fromAddress) {
+      setError(`Could not get from wallet address.`)
+      return
+    }
+
+    if (!toAddress) {
+      setError(`Could not get to wallet address.`)
+      return
+    }
+
+    const from: Asset = {
+      blockchain: fromToken.blockchain as string,
+      symbol: fromToken.symbol as string,
+      address: fromToken.address as string,
+    }
+    const to: Asset = {
+      blockchain: toToken.blockchain,
+      symbol: toToken.symbol as string,
+      address: toToken.address as string,
+    }
+    const amount: string = new BigNumber(inputAmount)
+      .shiftedBy(fromToken?.decimals as number)
+      .toString()
+
+    // const sampleImMessage = ethers.utils.defaultAbiCoder.encode(
+    //   ['(address,address)'],
+    //   [
+    //     [
+    //       fromToken.address || '0x0000000000000000000000000000000000000000',
+    //       userAddress,
+    //     ],
+    //   ]
+    // )
```

Comment:
We should pass it when test message passing is enabled in ui by user and tx type is EVM.

Reply:
I fixed it. I just was not sure if I should use 'fromAddress' or 'toAddress' instead of 'userAddress'. I used 'toAddress'.